File: app/src/main/java/com/privat/contacts/data/cloud/model/EmploymentNet.java
package com.privat.contacts.data.cloud.model;

import androidx.annotation.NonNull;

import com.google.gson.annotations.SerializedName;

import org.jetbrains.annotations.NotNull;

public class EmploymentNet {
    @NotNull
    private final String title;
    @SerializedName("key_skill")
    @NotNull
    private final String keySkill;

    public EmploymentNet(@NotNull String title, @NotNull String keySkill) {
        this.title = title;
        this.keySkill = keySkill;
    }

    public <T> T map(int userId, @NonNull Mapper<T> mapper) {
        return mapper.map(userId, title, keySkill);
    }

    public interface Mapper<T> {
        T map(int userId, @NotNull String title, @NotNull String keySkill);
    }
}

File: app/src/main/java/com/privat/contacts/data/cloud/model/EmploymentNet.kt
package com.privat.contacts.data.cloud.model

import com.google.gson.annotations.SerializedName

class EmploymentNet(private val title: String, @field:SerializedName("key_skill") private val keySkill: String) {
    fun <T> map(userId: Int, mapper: Mapper<T>): T {
        return mapper.map(userId, title, keySkill)
    }

    interface Mapper<T> {
        fun map(userId: Int, title: String, keySkill: String): T
    }
}
